### PR TITLE
Right click menu

### PR DESCRIPTION
**Pull request by gibsonliketheguitar**
#600 

Enable Right click on designated Field Types

### CODE REVIEW

**Comment by gibsonliketheguitar** [comment]
@shamsmosowi

**Comment by gibsonliketheguitar** [comment]
#630 New Branch, built on top of develop branch.

File: src/components/Table/TableRow.tsx [hunk 1]
```diff
@@ -8,9 +9,27 @@ export default function TableRow(props: RowRendererProps<any>) {
...
+  const { contextMenuRef }: any = useProjectContext();
+  function handleClick(e: any) {
+    e.preventDefault();
+    const input = e?.target as HTMLElement;
+    if (contextMenuRef?.current) {
+      contextMenuRef?.current?.setAnchorEl(input);
+    }
+  }
+  return <span onContextMenu={(e) => handleClick(e)}>{props.children}</span>;
+};
```
Comment (notsidney) [inline src/components/Table/TableRow.tsx:35]: I don’t think `ContextMenu` needs to be a separate component. I just set the `onContextMenu` prop on `Row` and the `e.target` is the cell `div`, which is the same in both methods.

File: src/components/Table/index.tsx [hunk 2]
```diff
@@ -262,6 +264,13 @@ export default function Table() {
                   });
                 }
               }}
+              onSelectedCellChange={({ rowIdx, idx }) => {
+                console.log("firing");
```
Comment (notsidney) [inline src/components/Table/index.tsx:268]: Remove console.log

File: src/components/fields/_BasicCell/BasicCellContextMenuActions.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        updateCell?.(selectedRow?.ref, selectedCol.key, paste);
+      } catch (error) {
+        //TODO check the coding style guide about error message
+        //Add breadcrumb handler her
+        console.log(error);
+      }
+    });
+
+    handleClose();
+  };
```
Comment (notsidney) [inline src/components/fields/_BasicCell/BasicCellContextMenuActions.tsx:57]: Just an FYI: we prefer to use the `async`/`await` syntax for Promises. For `navigator.clipboard`: https://web.dev/async-clipboard/

File: src/components/Table/ContextMenu/index.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export type ContextMenuRef = {
+  selectedCell: SelectedCell;
+  setSelectedCell: React.Dispatch<React.SetStateAction<SelectedCell | null>>;
+  anchorEl: HTMLElement | null;
+  setAnchorEl: React.Dispatch<
+    React.SetStateAction<PopoverProps["anchorEl"] | null>
+  >;
+};
```
Comment (notsidney) [inline src/components/Table/ContextMenu/index.tsx:22]: Side note: this state ref via context hack is a great candidate for migration to Jotai in the future

**Review by notsidney — APPROVED** [review]
I’ll make the small changes I mentioned in the comments, no need to do anything

File: src/components/Table/TableRow.tsx (hunk 1 above)
Comment (gibsonliketheguitar) [reply src/components/Table/TableRow.tsx:35]: Good catch, my dude! The less code the better. I will update the changes

**Pull request merged**